### PR TITLE
Support auto create sort dir in file sort mode, fix owner data race

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

When sort dir does not exist, try to create it recursively

### What is changed and how it works?

- check os.IsNotExist
- add a lock to owner access in server

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- No release note

### CODE REVIEW

**Comment by amyangfei** [comment]
/run-integration-tests tidb=release-4.0 tikv=release-4.0 pd=release-4.0

File: cdc/owner.go [hunk 1]
```diff
@@ -204,7 +205,14 @@ func (o *Owner) newChangeFeed(
 	if info.Engine == model.SortInFile {
 		err = util.IsDirAndWritable(info.SortDir)
 		if err != nil {
```
Comment (leoppro) [inline cdc/owner.go:207]: Could we try to create folders no matter what errors occur, such as no permissions, strange io errors, etc.?

File: cdc/http_handler.go [hunk 2]
```diff
@@ -65,8 +65,10 @@ func (s *Server) handleResignOwner(w http.ResponseWriter, req *http.Request) {
 		writeError(w, http.StatusBadRequest, errors.New("this api only supports POST method"))
 		return
 	}
+	s.ownerLock.RLock()
 	if s.owner == nil {
 		handleOwnerResp(w, concurrency.ErrElectionNoLeader)
```
Comment (leoppro) [inline cdc/http_handler.go:70]: why not put the operations of `lock\unlock` and checking nil into handleOwnerResp function

Comment (leoppro) [reply cdc/http_handler.go:70]: bad suggest, ignore me

File: cdc/owner.go (hunk 1 above)
Comment (amyangfei) [reply cdc/owner.go:211]: addressed

**Comment by amyangfei** [comment]
CI failure is caused by https://github.com/pingcap/ticdc/pull/816

**Pull request merged**